app/src/main/java/org/greenstand/android/TreeTracker/fragments/NewTreeFragment.kt

```diff
@@ -111,10 +116,15 @@ class NewTreeFragment :
             CameraHelper.takePictureForResult(this, selfie = false)
         } else {
             findNavController().popBackStack()
+            locationDataCapturer.turnOffTreeCaptureMode()
         }
     }
 
     override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+
+        newTreeUuid = locationDataCapturer.generatedTreeUuid
+        locationDataCapturer.turnOffTreeCaptureMode()
```

Comment:
We turn off the tree capture mode immediately on a activity result from the Camera screen. This is as per the spec as well to stop the ongoing Locations that are captured from getting tagged with the tree uuid generated in the prior 'Add Tree' flow. The cached `newTreeUuid` is retained and used to create the tree captured entity if the planter proceeds to save it.